cmd/podman/umount.go

```diff
@@ -24,18 +38,70 @@ func umountCmd(c *cli.Context) error {
 	}
 	defer runtime.Shutdown(false)
 
+	umountAll := c.Bool("all")
 	args := c.Args()
-	if len(args) == 0 {
+	if len(args) == 0 && !umountAll {
 		return errors.Errorf("container ID must be specified")
 	}
-	if len(args) > 1 {
-		return errors.Errorf("too many arguments specified")
+	if len(args) > 0 && umountAll {
+		return errors.Errorf("when using the --all switch, you may not pass any container IDs")
 	}
 
-	ctr, err := runtime.LookupContainer(args[0])
-	if err != nil {
-		return errors.Wrapf(err, "error looking up container %q", args[0])
+	umountContainerErrStr := "error unmounting container"
+	var lastError error
+	if len(args) > 0 {
+		for _, name := range args {
+			ctr, err := runtime.LookupContainer(name)
+			if err != nil {
+				if lastError != nil {
+					logrus.Error(lastError)
+				}
+				lastError = utils.WriteError(os.Stderr, errors.Wrapf(err, "%s %s", umountContainerErrStr, name), lastError)
+				continue
+			}
+
+			if err = unmountContainer(ctr); err != nil {
+				if lastError != nil {
+					logrus.Error(lastError)
+				}
+				lastError = utils.WriteError(os.Stderr, errors.Wrapf(err, "%s %q", umountContainerErrStr, name), lastError)
+				continue
+			}
+			fmt.Printf("%s\n", ctr.ID())
+		}
+	} else {
+		containers, err := runtime.GetContainers()
+		if err != nil {
+			return errors.Wrapf(err, "error reading Containers")
+		}
+		for _, ctr := range containers {
+			mountpoint, _ := ctr.Mountpoint()
+			ctrState, _ := ctr.State()
+			if ctrState == libpod.ContainerStateRunning || mountpoint == "" {
+				continue
+			}
+
+			if err = unmountContainer(ctr); err != nil {
+				if lastError != nil {
+					logrus.Error(lastError)
+				}
+				lastError = utils.WriteError(os.Stderr, errors.Wrapf(err, "%s %q", umountContainerErrStr, ctr.ID), lastError)
+				continue
+			}
+			fmt.Printf("%s\n", ctr.ID())
+		}
+	}
+	if lastError != nil {
+		logrus.Error(lastError)
```

Comment:
We don't need this, we print the error already when we return it

Reply:
I've removed this and the other logrus calls, just seemed silly to have the same error printed out twice with only a slight change in the first 10 or so chars between the two.